src/pymatgen/core/__init__.py

```diff
@@ -32,11 +32,11 @@
     pass
 
 
-SETTINGS_FILE = os.path.join(os.path.expanduser("~"), ".config", ".pmgrc.yaml")
-OLD_SETTINGS_FILE = os.path.join(os.path.expanduser("~"), ".pmgrc.yaml")
-MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
-PKG_DIR = os.path.dirname(MODULE_DIR)
-ROOT = os.path.dirname(PKG_DIR)
+SETTINGS_FILE: str = os.path.join(os.path.expanduser("~"), ".config", ".pmgrc.yaml")
+OLD_SETTINGS_FILE: str = os.path.join(os.path.expanduser("~"), ".pmgrc.yaml")
+MODULE_DIR: str = os.path.dirname(os.path.abspath(__file__))
+PKG_DIR: str = os.path.dirname(MODULE_DIR)
+ROOT: str = os.path.dirname(PKG_DIR)
```

Comment:
would there be a benefit to make them `Path` objects?

Reply:
so? what is your decision? I'm also fine with `str`, I just try to keep in mind to mae things as flexible and/or adjustable as possible. 😃

Reply:
Sorry for the lack of response I was busy fix the CI failures yesterday. I don't hold very strongly to prefer one over another at this moment (the decision by Janosh) and I might prefer to keep it as is for now until both have comparable performance maybe? At at that time we could globally migrate to `pathlib` handling?

Reply:
sounds good to me!

Reply:
Thanks for the confirm :)